### PR TITLE
Search: Group Search, Next, and Previous

### PR DESCRIPTION
**Pull request by kingthorin**
Add the existing Search, Next, and Previous items to a submenu (so that it's clear that Next and Prev apply to the search). Rename getters to be more clear. Set the hot key for "Prev" as ctrl+shift+g (next was already ctrl+g). Added/renamed resource keys for the submenu vs the search menu
item.

Add the new JMenu (group) via new ExtensionHookMenu convenience method(s) added to facilitate addition of sub-menu (JMenu) objects.

image

I believe this grouping clarifies things, and should fix zaproxy/zaproxy#5744

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
Should I also update these: https://github.com/zaproxy/zaproxy/blob/2b63145a4cd9d14187578f95b29463addd935d10/zap/src/main/resources/org/zaproxy/zap/resources/Messages.properties#L1978-L1981

To be `menu.edit.search.item.*`?

File: zap/src/main/java/org/parosproxy/paros/extension/ExtensionHookMenu.java [hunk 1]
```diff
@@ -245,6 +245,16 @@ public void addFileMenuItem(JMenuItem menuItem) {
         addFileMenuItemImpl(menuItem);
     }
 
+    /**
+     * Add a sub-menu to the File menu
+     *
+     * @param menu the sub-menu to add
+     * @since TODO Add version
+     */
+    public void addFileMenuItem(JMenu menu) {
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/extension/ExtensionHookMenu.java]: `MenuItem` → `Menu`?

Comment (kingthorin) [reply zap/src/main/java/org/parosproxy/paros/extension/ExtensionHookMenu.java]: I was worried about naming confusion, it isn't adding the File Menu. Maybe `addFileSubMenu`?

**Comment by kingthorin** [comment]
Tweaked

**Comment by thc202** [comment]
Thanks!

**Pull request merged**